typegate/engine/build.rs

```diff
@@ -1,9 +1,67 @@
 // Copyright Metatype OÜ, licensed under the Elastic License 2.0.
 // SPDX-License-Identifier: Elastic-2.0
 
-// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
-// SPDX-License-Identifier: MPL-2.0
+use std::path::PathBuf;
 
-fn main() -> shadow_rs::SdResult<()> {
-    shadow_rs::new()
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    shadow_rs::new()?;
+
```

Comment:
I believe we cover that in a standup but worth adding the reason why we use a build.rs instead of ghjk tasks?

Reply:
I was just doing a ghjk task at first but `build.rs` seemed more idiomatic since the pyrt.wasm file is always expected. Unlike `examples/templates`, it's a derived artifact.

So far, the compile/compression compile time bump hasn't been bad. Most of the typegate is in typescript so we rarely re-compile `typegate_engine`. We'll probably need to come up with a better scheme if we add more than one inline component.

Reply:
Also, in some scenarios, we pre-compile the module for other targets.

Reply:
In the short term, we will definitely only support Python but as soon as the wasm components are well supported in ruby and php, we likely going to add support for them.